GitUI/UserControls/GPGKeys/GPGSecretKeysCombobox.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.ComponentModel;
+using System.Diagnostics;
+using GitCommands.Git.Gpg;
+using GitCommands.Gpg;
+
+namespace GitUI.UserControls.GPGKeys
+{
+    public partial class GPGSecretKeysCombobox : GitModuleControl
+    {
+        private GPGKeysUIController _uiController;
+        private IGPGSecretKeysParser _secretKeysParser;
+        private bool _loadedKeys = false;
+        private string _selectedKeyID = "";
+
+        public IEnumerable<GpgKeyDisplayInfo> CurrentKeys { get => _loadedKeys ? (IEnumerable<GpgKeyDisplayInfo>)gpgKeyDisplayInfoBindingSource.DataSource : Enumerable.Empty<GpgKeyDisplayInfo>(); }
+
+        [Browsable(false)]
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+        public string KeyID
+        {
+            get => ((GpgKeyDisplayInfo)comboBoxKeys.SelectedItem)?.KeyID ?? "";
+            set
+            {
+                _selectedKeyID = value;
+                if (_loadedKeys)
+                {
+                    var empty = CurrentKeys?.First();
+                    var matched = CurrentKeys.FirstOrDefault(k => !string.IsNullOrWhiteSpace(value) && k.KeyID.EndsWith(value));
+                    comboBoxKeys.SelectedItem = matched ?? empty;
+                }
+            }
+        }
+
+        public bool IsDefaultSelected => _loadedKeys && SelectedIndex >= 0 && (comboBoxKeys.SelectedItem as GpgKeyDisplayInfo).IsDefault;
+        public GPGSecretKeysCombobox() : base()
+        {
+            InitializeComponent();
+            InitializeComplete();
+        }
+
+        private void InitControllers()
+        {
+            _secretKeysParser = new GPGSecretKeysParser();
```

Comment:
We should avoid newing up concrete implementations as that makes the testing story much more complicated. Absent of a DI, this should be passed in via `Bind()` or `Initialize` mechanisms as implemented in other parts of the codebase.

Reply:
Ok.  WIll look into.  Yes. DI would have helped a ton in this.